internal/trie/node/copy.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package node
+
+// Copy deep copies the branch.
+func (b *Branch) Copy() Node {
+	b.RLock()
+	defer b.RUnlock()
+
+	cpy := &Branch{
+		Key:        make([]byte, len(b.Key)),
+		Children:   b.Children, // copy interface pointers
```

Comment:
No, AFAIK we don't want to copy the entire child nodes, just the current node and re-use the pointers to the existing nodes. Note the pointers (aka addresses) are copied since `b.Children` is an array and not a slice.